src/components/BulkEnrollmentPage/CourseSearchResults.jsx

```diff
@@ -135,12 +159,19 @@ export const BaseCourseSearchResults = (props) => {
         columns={columns}
         data={searchResults?.hits || []}
         itemCount={searchResults?.nbHits}
+        isSelectable
+        isPaginated
+        manualSelectColumn={selectColumn}
+        onSelectedRowsChanged={onSelectedRowsChanged}
         SelectionStatusComponent={AddCoursesSelectionStatus}
         pageCount={searchResults?.nbPages || 1}
-        pageSize={searchResults?.hitsPerPage || 0}
-        selectedFlatRows={selectedCourses}
+        initialState={{
+          pageIndex: 0,
+          pageSize: NUM_CONTENT_ITEMS_PER_PAGE,
+          selectedRowIds: transformedSelectedRowIds,
+        }}
         initialTableOptions={{
-          getRowId: (row) => row.key,
+          getRowId: row => row.aggregation_key,
```

Comment:
[curious] do we want `row => row?.aggregation_key` just in case?

Reply:
@brobro10000 `row` _should_ never be undefined/null here. if it is, there's a bigger/separate issue.